name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/AccountBalanceChart.java

```diff
@@ -0,0 +1,91 @@
+package name.abuchen.portfolio.ui.views;
+
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.swt.widgets.Composite;
+import org.swtchart.ISeries;
+
+import name.abuchen.portfolio.model.Account;
+import name.abuchen.portfolio.model.AccountTransaction;
+import name.abuchen.portfolio.model.Transaction;
+import name.abuchen.portfolio.money.CurrencyConverter;
+import name.abuchen.portfolio.money.CurrencyConverterImpl;
+import name.abuchen.portfolio.money.ExchangeRateProviderFactory;
+import name.abuchen.portfolio.money.Values;
+import name.abuchen.portfolio.snapshot.AccountSnapshot;
+import name.abuchen.portfolio.ui.util.Colors;
+import name.abuchen.portfolio.ui.util.chart.TimelineChart;
+
+public class AccountBalanceChart extends TimelineChart{
+    
+
+    public AccountBalanceChart(Composite parent)
+    {
+        super(parent);
+        getTitle().setVisible(false);
+    }
+    
+    public void updateChart(Account account,ExchangeRateProviderFactory exchangeRateProviderFactory)
+    {
+        try
+        {
+            suspendUpdate(true);
+            
+            for (ISeries s : getSeriesSet().getSeries())
+                getSeriesSet().deleteSeries(s.getId());
+            
+            if (account == null)
+                return;
+            
+            List<AccountTransaction> tx = account.getTransactions();
+            
+            if (tx.isEmpty()) {
+                adjustRange();
+                suspendUpdate(false);
```

Comment:
Isn't the ```suspendUpdate(false)``` already part of the *finally* block? And if we need the ```adjustRange``` always, we might as well move it there as well.

I would take the liberty to quickly change that, squash your commits and merge it.